lib/Dialect/RTL/RTLOps.cpp

```diff
@@ -1315,6 +1315,172 @@ void ReadInOutOp::build(OpBuilder &builder, OperationState &result,
+
+  auto actStructType = declStructType.dyn_cast<StructType>();
+  if (!actStructType) {
+    parser.emitError(loc, "expected struct type");
+    return failure();
+  }
+
+  llvm::SmallVector<Type, 4> structInnerTypes;
+  actStructType.getInnerTypes(structInnerTypes);
+  result.addTypes(actStructType);
+
+  if (parser.resolveOperands(operands, structInnerTypes, loc, result.operands))
+    return failure();
+  return success();
+}
+
+static void print(OpAsmPrinter &printer, rtl::StructCreateOp op) {
+  printer << op.getOperationName() << " (";
+  printer.printOperands(op.input());
+  printer << ")";
+  printer.printOptionalAttrDict(op.getAttrs());
+  printer << " : " << op.getType();
+}
+
+//===----------------------------------------------------------------------===//
+// StructExplodeOp
+//===----------------------------------------------------------------------===//
+
+static ParseResult parseStructExplodeOp(OpAsmParser &parser,
+                                        OperationState &result) {
+  OpAsmParser::OperandType operand;
+  Type declStructType;
+
+  auto loc = parser.getCurrentLocation();
+
+  if (parser.parseOperand(operand) ||
+      parser.parseOptionalAttrDict(result.attributes) ||
+      parser.parseColonType(declStructType))
+    return failure();
+
+  auto actStructType = declStructType.dyn_cast<StructType>();
+  if (!actStructType) {
+    parser.emitError(loc, "expected struct type");
+    return failure();
+  }
+
+  llvm::SmallVector<Type, 4> structInnerTypes;
+  actStructType.getInnerTypes(structInnerTypes);
+  result.addTypes(structInnerTypes);
+
+  if (parser.resolveOperand(operand, actStructType, result.operands))
+    return failure();
+  return success();
+}
+
+static void print(OpAsmPrinter &printer, rtl::StructExplodeOp op) {
+  printer << op.getOperationName() << " ";
+  printer.printOperand(op.input());
+  printer.printOptionalAttrDict(op.getAttrs());
+  printer << " : " << op.input().getType();
+}
+
+//===----------------------------------------------------------------------===//
+// StructExtractOp
+//===----------------------------------------------------------------------===//
+
+static ParseResult parseStructExtractOp(OpAsmParser &parser,
+                                        OperationState &result) {
+  OpAsmParser::OperandType operand;
+  StringAttr fieldName;
+  Type declStructType;
+
+  auto loc = parser.getCurrentLocation();
+
+  if (parser.parseOperand(operand) || parser.parseLSquare() ||
+      parser.parseAttribute(fieldName, "field", result.attributes) ||
+      parser.parseRSquare() ||
+      parser.parseOptionalAttrDict(result.attributes) ||
+      parser.parseColonType(declStructType))
+    return failure();
+
+  auto actStructType = declStructType.dyn_cast<StructType>();
+  if (!actStructType) {
+    parser.emitError(loc, "expected struct type");
+    return failure();
+  }
+
+  Type resultType = actStructType.getFieldType(fieldName.getValue());
+  if (!resultType)
+    return failure();
```

Comment:
This needs to emit an error.  Also I think that errors implicitly convert to ParseResults, so you can just do: `return parser.emitError(...`